drivers/overlay/overlay.go

```diff
@@ -364,12 +364,14 @@ func Init(home string, options graphdriver.Options) (graphdriver.Driver, error)
 		// Try to enable project quota support over xfs.
 		if d.quotaCtl, err = quota.NewControl(home); err == nil {
 			projectQuotaSupported = true
-		} else if opts.quota.Size > 0 {
-			return nil, fmt.Errorf("Storage option overlay.size not supported. Filesystem does not support Project Quota: %v", err)
+		} else {
+			if opts.quota.Size > 0 || opts.quota.Inodes > 0 {
+				return nil, fmt.Errorf("Storage option overlay.size and overlay.inodes not supported. Filesystem does not support Project Quota: %v", err)
+			}
 		}
-	} else if opts.quota.Size > 0 {
+	} else if opts.quota.Size > 0 || opts.quota.Inodes > 0 {
 		// if xfs is not the backing fs then error out if the storage-opt overlay.size is used.
-		return nil, fmt.Errorf("Storage option overlay.size only supported for backingFS XFS. Found %v", backingFs)
+		return nil, fmt.Errorf("Storage option overlay.size and overlay.inodes only supported for backingFS XFS. Found %v", backingFs)
```

Comment:
Based on a quick look at kernel sources, ext4 should also support project quotas.

Reply:
I believe their is more work required to make ext4 work, which I am not doing in this PR.

// TODO: use generic quota control ioctl FS_IOC_FS{GET,SET}XATTR
//       for both xfs/ext4 for kernel version >= v4.5